src/main/scala/strawman/collection/Set.scala

```diff
@@ -42,9 +58,13 @@ trait SetLike[A, +C[X] <: Set[X]]
 trait SetMonoTransforms[A, +Repr]
   extends IterableMonoTransforms[A, Repr] {
 
-  def & (that: Set[A]): Repr
+  def & (that: Set[A]): Repr = this.filter(that)
+
+}
+
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
 
-  def ++ (that: Set[A]): Repr
+  def ++ (that: Set[A]): C[A]
```

Comment:
Note that I moved `++` into a new `SetPolyTransforms` trait despite the method returns a `Set` with the same element type `A`. This is just in order to be able to provide a default implementation (see in `immutable.Set`) that gets the self-type right.